File: src/data_structure_stack_queue/ArrayStack.java
package data_structure_stack_queue;

/**
 * Created by parkdongjo on 2017. 6. 25..
 */
public class ArrayStack {

    private final int ARR_MAX_SIZE = 10;

    private Integer[] array = new Integer[ARR_MAX_SIZE];
    private int size;


    public ArrayStack() {
        this.size = 0;
    }

    public int top() {
        int result = 1;

        if(isEmpty()) {
            result = -1;
        }else{
            result = array[size-1];
        }
        return result;
    }

    public int push(int data) {

        if (!isFull()) {
            array[size] = data;
            return 1;
        }
        return -1;
    }

    public int pop() {
        int result = 1;

        if(isEmpty()) {
            result = array[size-1];
        }else {
            result = -1;
        }

        return result;
    }

    public boolean isFull() {
        boolean result = true;

        if(size == ARR_MAX_SIZE) {
            result = false;
        }
        return result;
    }

    public boolean isEmpty() {
        boolean result = true;

        if(size != 0) {
            result = false;
        }

        return result;
    }

}

File: src/data_algorithm_test/rank_algorithm/ArrayStack.kt
package data_algorithm_test.rank_algorithm

class ArrayStack {
    private val MAX_SIZE = 100
    private var pArray: IntArray = IntArray(MAX_SIZE, {i -> -1})
    private var pSize: Int = 0

    var size: Int
        set(size) {
            this.pSize = size
        }
        get() = this.pSize

    fun top(): Int {
        if (isEmpty()) {
            return -1
        } else {
            return pArray[this.size-1]
        }
    }

    fun pop(): Int {
        if (isEmpty()) {
            return -1
        } else {
            return pArray[this.size-1].also { this.size-- }
        }
    }

    fun push(data: Int) {
        if (this.size == MAX_SIZE) {
            throw ArrayIndexOutOfBoundsException()
        } else {
            this.pArray[this.size++] = data
        }
    }

    fun isEmpty(): Boolean {
        return this.size == 0
    }

}